### PR TITLE
Enable compiler driven save actions

### PR DESCRIPTION
**Pull request by sschaef**
See commit messages for more information

### CODE REVIEW

File: org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/SaveActionExtensions.scala [hunk 1]
```diff
@@ -153,31 +147,63 @@ trait SaveActionExtensions extends HasLogger {
...
-        performExtension(instance, udoc) {
-          instance.global.asInstanceOf[IScalaPresentationCompiler].asyncExec {
-            instance.perform()
-          }.getOrElse(Seq())()
+    val timeout = saveActionTimeout
+
+    def loop(xs: Seq[(SaveActionSetting, CompilerSupportCreator)]): Unit = xs match {
+      case Nil ⇒
+
+      case (setting, ext) :: xs if isEnabled(setting.id) ⇒
```
Comment (Kwestor) [inline org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/SaveActionExtensions.scala]: `xs` is a `Seq`, but `::` matches only if underlying implementation is a `List`. Using `+:` is more safe here.

File: org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/SaveActionExtensions.scala [hunk 2]
```diff
@@ -153,31 +147,63 @@ trait SaveActionExtensions extends HasLogger {
...
+        val res = FutureUtils.performWithTimeout(timeout) {
+          EclipseUtils.withSafeRunner(s"An error occurred while executing compiler driven save action.") {
```
Comment (Kwestor) [inline org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/SaveActionExtensions.scala]: Is there a reason not to include `${setting.id}` here?

File: org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/SaveActionExtensions.scala [hunk 3]
```diff
@@ -153,31 +147,63 @@ trait SaveActionExtensions extends HasLogger {
...
+               | complete in time, no further save actions are executed. Please consider
+               | to disable save actions that rely on the compiler. The performed save
+               | action itself couldn't be aborted, therefore if you know that it may
+               | never complete in future, you may wish to restart your Eclipse to
+               | clean up your VM.
+               |
+               |""".stripMargin.replaceAll("\n", ""))
+        }
+
+      case _ :: xs ⇒
```
Comment (Kwestor) [inline org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/SaveActionExtensions.scala]: Same as above, `::` matches only lists. Also, `xs` shadows `xs` method argument, maybe rename it to `tail` (same above)?

**Comment by Kwestor** [comment]
Review done - looks good, see my comments for some nitpicking.

File: org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/SaveActionExtensions.scala (hunk 3 above)
Comment (sschaef) [inline org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/SaveActionExtensions.scala]: The shadowing is on purpose actually.

**Comment by sschaef** [comment]
Thanks for the review, I took care of it.

**Comment by Kwestor** [comment]
LGTM then.

**Pull request merged**